### ISSUE TITLE
Refactor defaulting logic to ensure  completeness of the function-managed templates 

### ISSUE THREAD

**Issue by kwiatekus** [kyma-project/kyma#15499]
**Description**

Modify the deafulting logic to ensure spec completeness in the area of the templates of function managed deployment and build job. Assume precedence of fields specifically set for a given template. The lower specification fields must have higher precedence over the high-level fields. This allows for overriding a limited set of template subfields without filling out the template completely.

 - preserve the high-level specification fields defaulting using the defaulting webhook.
 - If the template fields are not set, they should be defaulted based on the high-level specification by the Function controller.
 - If the template fields are partially set, the controller must respect the user-defined values and default unset subfields based on the high-level fields.

**Reasons**

We want the resource templates to be the source of truth for the controller when scheduling runtime pod and build job.
However, we dont want to force the user to always fill them up in the spec as they are considered a way to fine-tune the resources in more demanding case. Therefore we need to have a defaulting login in place that makes it easy to fill up function spec for the most cases (or beginner user) and assures completeness of the spec.

**Attachments**
[proposal](https://github.com/kyma-project/kyma/blob/main/components/function-controller/design/subresources_templates_proposal.md)

**Comment by kwiatekus** [comment]
Depends on #15402

**Comment by kwiatekus** [comment]
Closing in favor of https://github.com/kyma-project/kyma/issues/15737, https://github.com/kyma-project/kyma/issues/14998, https://github.com/kyma-project/kyma/issues/14877
